src/main/java/com/vella/blog/service/BlogServiceImpl.java

```diff
@@ -110,10 +110,10 @@ public Blog saveBlog(BlogCreateRequest request, String token) {
       blog.setDateEditedAt(LocalDate.now());
       blog.setTimeEditedAt(LocalTime.now());
 
-      if(tokenRepo.findUserByToken(token).isEmpty()){
+      if(userRepo.findUserByToken(token.substring(7)).isEmpty()){
```

Comment:
The expression in the if block is a bit unreadable. A better way of writing this would be to extract it to it's own boolean variable, e.g.:

```suggestion
String tokenSubstring = token.substring(7);
Optional<User> userOptional = userRepo.findUserByToken(tokenSubstring);

if (userOptional.isEmpty()) {
    throw new IllegalArgumentException("Error saving new blog. No AppUser found.");
}

User user = userOptional.get();

```

This way you also avoid unnecessarily calling the same repo method twice. This means two queries to the database asking for the same data. It is not performant and can slow down your app. Any unnecessary calls to the database should always be avoided.

Reply:
Fix this issue anywhere else it occurs.